From 5ceef155c7de21d521ea87b352498c39c678465d Mon Sep 17 00:00:00 2001
From: Sebastian Krzyszkowiak <dos@dosowisko.net>
Date: Tue, 26 Feb 2019 04:40:38 +0100
Subject: [PATCH 3/3] SDL: fix unit in touch event's timestamps

---
 src/sdl/sdl_touch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sdl/sdl_touch.c b/src/sdl/sdl_touch.c
index e5912c76f..6afde2cac 100644
--- a/src/sdl/sdl_touch.c
+++ b/src/sdl/sdl_touch.c
@@ -167,7 +167,7 @@ void _al_sdl_touch_input_event(SDL_Event *e)
       return;
    }
 
-   generate_touch_input_event(type, e->tfinger.timestamp, touchId,
+   generate_touch_input_event(type, e->tfinger.timestamp / 1000.0, touchId,
                               touch_input->state.touches[touchId].x,
                               touch_input->state.touches[touchId].y,
                               touch_input->state.touches[touchId].dx,
-- 
2.21.0

